### PR TITLE
Cabol.first draft

### PR DESCRIPTION
**Pull request by cabol**
Riak Extras 1st draft
- Functions `find_by/3` and `find_by/5` were overwritten to support complex/composed data structures.

### CODE REVIEW

File: test/sumo_basic_SUITE.erl [hunk 1]
```diff
@@ -0,0 +1,137 @@
+-module(sumo_basic_SUITE).
```
Comment (gadgetci) [inline test/sumo_basic_SUITE.erl]: According to **Elvis**:
> The module "sumo_basic_SUITE" does not respect the format defined by the regular expression '"^([a-z][a-z0-9]*_?)*$"'.

File: test/sumo_test_purchase_order.erl [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  #{
+    part_num      => maps:get(part_num, Item),
+    product_name  => maps:get(product_name, Item),
+    quantity      => to_int(maps:get(quantity, Item)),
+    unit_price    => to_int(maps:get(unit_price, Item)),
+    price         => to_int(maps:get(price, Item))
+  }.
+
+%% @private
+wakeup_items(Items) ->
```
Comment (elbrujohalcon) [inline test/sumo_test_purchase_order.erl]: Can't this function be just `[wakeup_item(V) || V <- maps:values(Items)]` ?

Comment (elbrujohalcon) [inline test/sumo_test_purchase_order.erl]: Or `lists:map(fun wakeup_item/1, maps:values(Items))` ?

File: src/sumo_store_riak_ext.erl [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% Private API.
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+%% @private
+%% @doc Search all docs that match with the given query, but only keys are
+%%      returned.
+%%      IMPORTANT: assumes that default schema 'yokozuna' is being used.
+search_keys_by(Conn, Index, Query, Limit, Offset) ->
```
Comment (gadgetci) [inline src/sumo_store_riak_ext.erl]: According to **Xref**:
> `sumo_store_riak_ext:search_keys_by/5` calls undefined function `sumo_store_riak:search/5`

File: src/sumo_store_riak_ext.erl [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  find_by(DocName, Conditions, 0, 0, State).
+
+-spec find_by(
+  sumo:schema_name(),
+  sumo:conditions(),
+  non_neg_integer(),
+  non_neg_integer(),
+  state()
+) -> sumo_store:result([sumo_internal:doc()], state()).
+find_by(DocName,
```
Comment (gadgetci) [inline src/sumo_store_riak_ext.erl]: According to **Xref**:
> `sumo_store_riak_ext:find_by/5` calls undefined function `sumo_store_riak:fetch_docs/5`

Comment (gadgetci) [inline src/sumo_store_riak_ext.erl]: According to **Xref**:
> `sumo_store_riak_ext:find_by/5` calls undefined function `sumo_store_riak:fetch_map/4`

Comment (gadgetci) [inline src/sumo_store_riak_ext.erl]: According to **Xref**:
> `sumo_store_riak_ext:find_by/5` calls undefined function `sumo_store_riak:build_query/1`

Comment (gadgetci) [inline src/sumo_store_riak_ext.erl]: According to **Xref**:
> `sumo_store_riak_ext:find_by/5` calls undefined function `sumo_store_riak:rmap_to_doc/2`

**Pull request merged**